vector/src/main/java/fr/gouv/tchap/util/DinsicUtils.java

```diff
@@ -997,20 +999,66 @@ public static String getRoomAccessRule(@NonNull Room room) {
         return rule;
     }
 
-    public static void setRoomAccessRule (MXSession session, Room room, @NonNull String rule, final ApiCallback<Void> callback) {
+    public static void setRoomAccessRule(MXSession session, Room room, @NonNull String rule, final ApiCallback<Void> callback) {
 
         Map<String, Object> content = new HashMap<>();
         content.put(RoomAccessRulesKt.STATE_EVENT_CONTENT_KEY_RULE, rule);
 
-        session.getRoomsApiClient().sendStateEvent(room.getRoomId(), RoomAccessRulesKt.STATE_EVENT_TYPE, "", content, callback);
+        session.getRoomsApiClient().sendStateEvent(room.getRoomId(), RoomAccessRulesKt.EVENT_TYPE_STATE_ROOM_ACCESS_RULES, "", content, callback);
+    }
+
+    /**
+     * Get the current room retention period in days.
+     *
+     * @param room the room.
+     * @return the room retention period
+     */
+    @NonNull
+    public static int getRoomRetention(@NonNull Room room) {
```

Comment:
You should create a new file for DinsicUtils in kotlin, you will write nicer code :)

Reply:
done